rskj-core/src/main/java/org/ethereum/core/BlockHeader.java

```diff
@@ -471,7 +498,30 @@ public String getParentShortHash() {
         return HashUtil.shortHash(getParentHash().getBytes());
     }
 
-    private static BigInteger parseBigInteger(byte[] bytes) {
-        return bytes == null ? BigInteger.ZERO : BigIntegers.fromUnsignedByteArray(bytes);
+    public byte[] getMiningForkDetectionData() {
+        if(includeForkDetectionData) {
+            if (hasMiningFields()) {
+                byte[] encodedBlock = getEncoded(false, false);
+                byte[] hashForMergedMining = HashUtil.keccak256(encodedBlock);
+
+                byte[] hashForMergedMiningPrefix = Arrays.copyOfRange(hashForMergedMining, 0, HASH_FOR_MERGED_MINING_PREFIX_LENGTH);
+                byte[] coinbaseTransaction = getBitcoinMergedMiningCoinbaseTransaction();
+
+                List<Byte> hashForMergedMiningPrefixAsList = Arrays.asList(ArrayUtils.toObject(hashForMergedMiningPrefix));
+                List<Byte> coinbaseAsList = Arrays.asList(ArrayUtils.toObject(coinbaseTransaction));
+
+                int position = Collections.lastIndexOfSubList(coinbaseAsList, hashForMergedMiningPrefixAsList);
+                if (position == -1) {
+                    return new byte[12];
```

Comment:
Sometimes this method returns new byte[0] and some times new byte[12] (zero-filled). What is the reason for not returning always the same? Does the calling code distinguished between these two values? If these are error conditions, wouldn't be better to rise an exception ?

Reply:
The case of `new byte[12]` is an error condition, I changed it to throw an exception. 
It is ok to return `new byte[0]` because its consistent with the rest of the code for the case when that value can not be calculated.